### PR TITLE
PE-2814: Snapshot to be created

### PR DESCRIPTION
**Pull request by matibat**
Changes
- Makes SnapshotItemToBeCreated to no longer extend SnapshotItem
- SnapshotItemToBeCreated has the responsibility of knowing the data of the snapshot
- Updates the tests for SnapshotItem
- Implements the tests for SnapshotItemToBeCreated

--- Releases ---
Android release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:android:6cf0cd5ec064fad3ffce07/releases/2s12d12f3cj40
iOS release: https://appdistribution.firebase.google.com/testerapps/1:305132849030:ios:06ea33a95a2e0d42ffce07/releases/5qb0i8mp9dbp0

### CODE REVIEW

File: lib/utils/snapshots/snapshot_item_to_be_created.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    final tags = node.tags;
+    final entityTypeTags =
+        tags.where((tag) => tag.name == EntityTag.entityType);
+
+    return entityTypeTags.any((tag) => tag.value == 'snapshot');
+  }
+
+  String _jsonMetadataOfTxId(String txId) {
+    // FIXME: let me read the DB, and go to the network if not present.
+    return "TODO";
```
Comment (matibat) [inline lib/utils/snapshots/snapshot_item_to_be_created.dart:54]: Fix it 😆

File: lib/utils/snapshots/snapshot_item_to_be_created.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    return snapshotDataStream;
+  }
+
+  bool _isSnapshotTx(DriveHistoryTransaction node) {
+    final tags = node.tags;
+    final entityTypeTags =
+        tags.where((tag) => tag.name == EntityTag.entityType);
+
+    return entityTypeTags.any((tag) => tag.value == 'snapshot');
```
Comment (matibat) [inline lib/utils/snapshots/snapshot_item_to_be_created.dart:49]: Use a constant instead?

File: lib/utils/snapshots/snapshot_item_to_be_created.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    required this.driveId,
+    required this.subRanges,
+    required this.source,
+  });
+
+  Stream<Uint8List> getSnapshotData() {
+    final txSnapshotStream = source.map<TxSnapshot>(
+      (node) => TxSnapshot(
+        gqlNode: node,
+        jsonMetadata: _isSnapshotTx(node) ? '' : _jsonMetadataOfTxId(node.id),
```
Comment (matibat) [inline lib/utils/snapshots/snapshot_item_to_be_created.dart:34]: This conditional ensures we won't nest snapshots' data. See the test case `"the returned data won't contain metadatadata of other snapshots, but only the gql node"`.

File: test/utils/snapshot_item_to_be_created_test.dart [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            .join();
+
+        expect(
+          snapshotData,
+          '{"txSnapshots":[{"gqlNode":{"id":"tx-8","owner":{"address":"1234567890"},"bundledIn":{"id":"ASDASDASDASDASDASD"},"block":{"height":8,"timestamp":800},"tags":[]},"jsonMetadata":"TODO"}]}',
+        );
+      });
+
+      test(
+          'the returned data won\'t contain metadatadata of other snapshots, but only the gql node',
```
Comment (matibat) [inline test/utils/snapshot_item_to_be_created_test.dart:54]:
```suggestion
          'the returned data won\'t contain the json metadata of other snapshots, but only the gql node',
```

File: lib/utils/snapshots/snapshot_item_to_be_created.dart (hunk 1 above)
Comment (matibat) [reply lib/utils/snapshots/snapshot_item_to_be_created.dart]: [cc07ce8](https://github.com/ardriveapp/ardrive-web/pull/846/commits/cc07ce8026f8cf4e07f86056469128ed52542f70)

File: lib/utils/snapshots/snapshot_item_to_be_created.dart (hunk 2 above)
Comment (matibat) [reply lib/utils/snapshots/snapshot_item_to_be_created.dart:53]: Where should it be at?

Comment (matibat) [reply lib/utils/snapshots/snapshot_item_to_be_created.dart:53]: What about creating an EntityType class with the value for snapshots? Although we don't have more entity type values in that source file...
`lib/entities/constants.dart`

Comment (karlprieb) [reply lib/utils/snapshots/snapshot_item_to_be_created.dart:53]: We do have https://github.com/ardriveapp/ardrive-web/blob/bddd471bd922b8024bb6888bb9e595bef393da5c/lib/entities/constants.dart#L35-L39 👍 
I think we can use it for snapshots as well :D

File: lib/utils/snapshots/snapshot_item_to_be_created.dart [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+import 'package:ardrive/entities/entities.dart';
+import 'package:ardrive/entities/string_types.dart';
+import 'package:ardrive/services/arweave/graphql/graphql_api.graphql.dart';
+import 'package:ardrive/utils/snapshots/snapshot_types.dart';
+import 'package:ardrive/utils/snapshots/tx_snapshot_to_snapshot_data.dart';
+
+import 'height_range.dart';
+
+typedef DriveHistoryTransaction
```
Comment (karlprieb) [inline lib/utils/snapshots/snapshot_item_to_be_created.dart:12]: 👍 thank you :D

File: lib/utils/snapshots/snapshot_item_to_be_created.dart (hunk 2 above)
Comment (matibat) [reply lib/utils/snapshots/snapshot_item_to_be_created.dart]: Sleepy Mati did't see that yesterday :sleeping: 

Applied - 951cf79bd06432c7960e21e653c731b2f099f8f5

**Pull request merged**